[
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:04:32,093 INFO cards.py took 0:04:27.404738\n```",
        "ts": "1555401833.041300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "Nightly completed",
        "ts": "1555401903.041400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:05:42,195 INFO nightly.py took 4:18:23.335196\n```",
        "ts": "1555401903.041500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:08:12,863 INFO backup.py took 0:08:10.953472\n```",
        "ts": "1555402054.041600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "",
        "ts": "1555402318.041700",
        "username": "incoming-webhook",
        "bot_id": "B5XN109SM",
        "attachments": [
            {
                "fallback": "<@channel>   *[Amazon CloudWatch Notification]* \nSubject: ALARM: \"EDHREC NetworkOut\" in US East (N. Virginia)\nMessage: {\"AlarmName\":\"EDHREC NetworkOut\",\"AlarmDescription\":null,\"AWSAccountId\":\"083767677168\",\"NewStateValue\":\"ALARM\",\"NewStateReason\":\"Threshold Crossed: 1 datapoint [6.497260622E8 (16\/04\/19 08:06:00)] was greater than or equal to the threshold (3.0E8).\",\"StateChangeTime\":\"2019-04-16T08:11:57.677+0000\",\"Region\":\"US East (N. Virginia)\",\"OldStateValue\":\"OK\",\"Trigger\":{\"MetricName\":\"NetworkOut\",\"Namespace\":\"AWS\/EC2\",\"StatisticType\":\"Statistic\",\"Statistic\":\"AVERAGE\",\"Unit\":null,\"Dimensions\":[{\"value\":\"i-0f7f3750e584e120e\",\"name\":\"InstanceId\"}],\"Period\":300,\"EvaluationPeriods\":1,\"ComparisonOperator\":\"GreaterThanOrEqualToThreshold\",\"Threshold\":3.0E8,\"TreatMissingData\":\"\",\"EvaluateLowSampleCountPercentile\":\"\"}}\nTimestamp: 2019-04-16T08:11:57.710Z ",
                "text": "<@channel>   *[Amazon CloudWatch Notification]* \nSubject: ALARM: \"EDHREC NetworkOut\" in US East (N. Virginia)\nMessage: {\"AlarmName\":\"EDHREC NetworkOut\",\"AlarmDescription\":null,\"AWSAccountId\":\"083767677168\",\"NewStateValue\":\"ALARM\",\"NewStateReason\":\"Threshold Crossed: 1 datapoint [6.497260622E8 (16\/04\/19 08:06:00)] was greater than or equal to the threshold (3.0E8).\",\"StateChangeTime\":\"2019-04-16T08:11:57.677+0000\",\"Region\":\"US East (N. Virginia)\",\"OldStateValue\":\"OK\",\"Trigger\":{\"MetricName\":\"NetworkOut\",\"Namespace\":\"AWS\/EC2\",\"StatisticType\":\"Statistic\",\"Statistic\":\"AVERAGE\",\"Unit\":null,\"Dimensions\":[{\"value\":\"i-0f7f3750e584e120e\",\"name\":\"InstanceId\"}],\"Period\":300,\"EvaluationPeriods\":1,\"ComparisonOperator\":\"GreaterThanOrEqualToThreshold\",\"Threshold\":3.0E8,\"TreatMissingData\":\"\",\"EvaluateLowSampleCountPercentile\":\"\"}}\nTimestamp: 2019-04-16T08:11:57.710Z ",
                "id": 1,
                "color": "F35A00",
                "fields": [
                    {
                        "title": "Severity",
                        "value": "ALARM",
                        "short": true
                    }
                ],
                "mrkdwn_in": [
                    "text"
                ]
            }
        ]
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:39:06,150 INFO archidekt.py took 0:29:55.256040\n```",
        "ts": "1555403907.041800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:42:58,551 WARNING Chaos Orb from Ugin, the Ineffable in <https:\/\/deckstats.net\/decks\/44038\/1262728-rona-midrange\/en>\n```",
        "ts": "1555404140.041900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:51:48,429 WARNING Chaos Orb from Ugin, the Ineffable in <https:\/\/deckstats.net\/decks\/55029\/1057448-exquisite-invention\/en>\n```",
        "ts": "1555404670.042000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:54:34,473 WARNING Chaos Orb from Ashiok, Dream Render in <https:\/\/deckstats.net\/decks\/63287\/1264962-dragon-reborn-wheel-of-time\/en>\n```",
        "ts": "1555404836.042100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:57:35,600 WARNING Chaos Orb from Ugin, the Ineffable in <https:\/\/deckstats.net\/decks\/120705\/1264624-assassin-army\/en>\n```",
        "ts": "1555405017.042200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 08:58:44,711 WARNING Chaos Orb from Nicol Bolas, Dragon-God in <https:\/\/deckstats.net\/decks\/84112\/1264818-dragons-reborn\/en>\n```",
        "ts": "1555405086.042300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 09:02:02,501 WARNING Chaos Orb from Ugin, the Ineffible in <https:\/\/deckstats.net\/decks\/4441\/1233525-commander-colorless-kozilek-th\/en>\n```",
        "ts": "1555405284.042400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 09:08:00,273 WARNING Chaos Orb from Nicol Bolas, Dragon-God in <https:\/\/deckstats.net\/decks\/21457\/1051138-progenitus-\/en>\n```",
        "ts": "1555405642.042500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 09:10:57,271 INFO deckstats.py took 0:29:51.224445\n```",
        "ts": "1555405818.042600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 10:02:13,518 INFO mtggoldfish.py took 0:30:39.827368\n```",
        "ts": "1555408895.042700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 10:26:39,118 INFO update_deckswithprices.py took 0:24:22.974244\n```",
        "ts": "1555410366.042800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```[16\/Apr\/2019:10:26:42] HTTP \nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/web\/dynamic.py\", line 81, in index\n    tribelinks=data['tribelinks'])\n  File \"\/home\/ubuntu\/edhrec\/web\/web.py\", line 210, in make_page\n    } for s in cards.get_sets()[:5]\n  File \"\/home\/ubuntu\/edhrec\/core\/cards.py\", line 473, in get_sets\n    return cache.get('cards\/sets')\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 52, in get\n    cur.execute(query, (key.lower(),))\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/psycopg2\/extras.py\", line 141, in execute\n    return super(DictCursor, self).execute(query, vars)\npsycopg2.OperationalError: out of memory\nDETAIL:  Failed on request of size 114119813.\n\n[16\/Apr\/2019:10:26:44] HTTP \nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 23, in _getcursor\n    yield con.cursor(cursor_factory=psycopg2.extras.DictCursor)\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 52, in get\n    cur.execute(query, (key.lower(),))\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/psycopg2\/extras.py\", line 141, in execute\n    return super(DictCursor, self).execute(query, vars)\npsycopg2.OperationalError: lost synchronization with server: got message type \"D\", length 114119859\n\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 630, in respond\n    self._do_respond(path_info)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cprequest.py\", line 689, in _do_respond\n    response.body = self.handler()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/lib\/encoding.py\", line 221, in __call__\n    self.body = self.oldhandler(*args, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/cherrypy\/_cpdispatch.py\", line 54, in __call__\n    return self.callable(*self.args, **self.kwargs)\n  File \"\/home\/ubuntu\/edhrec\/web\/dynamic.py\", line 104, in index\n    return make_deck(kwargs['cardhash'])\n  File \"\/home\/ubuntu\/edhrec\/web\/dynamic.py\", line 443, in make_deck\n    dynamic=True\n  File \"\/home\/ubuntu\/edhrec\/web\/web.py\", line 210, in make_page\n    } for s in cards.get_sets()[:5]\n  File \"\/home\/ubuntu\/edhrec\/core\/cards.py\", line 473, in get_sets\n    return cache.get('cards\/sets')\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 67, in get\n    raise NotInCache(key)\n  File \"\/usr\/lib\/python3.5\/contextlib.py\", line 77, in __exit__\n    self.gen.throw(type, value, traceback)\n  File \"\/home\/ubuntu\/edhrec\/core\/cache.py\", line 25, in _getcursor\n    con.commit()\npsycopg2.InterfaceError: connection already closed\n```",
        "ts": "1555410366.042900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 10:26:53,003 INFO New deck count from past 2 days: \n- MTGGoldfish: 356\n- deckstats: 368\n- Other: 0\n- Archidekt: 535\n```",
        "ts": "1555410374.043000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "Nightly started, onlydecks=False, skipdecks=False",
        "ts": "1555410384.043100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 12:47:48,985 INFO New Trending 2019-04-16\n{\"name\": \"Sandwurm Convergence\", \"s\": [0, 12, 8, 10, 17, 22, 19], \"z\": 1.2054793179390164}\n{\"name\": \"Seal of Primordium\", \"s\": [0, 7, 8, 7, 14, 13, 14], \"z\": 1.1672324746892049}\n{\"name\": \"Khalni Garden\", \"s\": [0, 6, 9, 8, 13, 14, 16], \"z\": 1.1622204236000295}\n{\"name\": \"Memory Lapse\", \"s\": [0, 4, 5, 5, 8, 11, 10], \"z\": 1.132547774748432}\n{\"name\": \"Fall of the Titans\", \"s\": [0, 2, 3, 3, 6, 7, 6], \"z\": 1.1162575232452048}\n{\"name\": \"Sundering Growth\", \"s\": [0, 7, 4, 9, 14, 10, 13], \"z\": 1.1147353640199462}\n{\"name\": \"Lifeblood Hydra\", \"s\": [0, 3, 7, 5, 11, 9, 13], \"z\": 1.1105551676199057}\n{\"name\": \"Relic of Progenitus\", \"s\": [0, 8, 5, 9, 11, 14, 13], \"z\": 1.1089916579427712}\n{\"name\": \"Temur Sabertooth\", \"s\": [0, 17, 25, 21, 29, 34, 29], \"z\": 1.1074782664675162}\n{\"name\": \"Conflux\", \"s\": [0, 4, 3, 9, 16, 11, 9], \"z\": 1.1054693527458208}\n2019-04-16 12:47:49,000 INFO daily.py took 4:47:44.497672\n```",
        "ts": "1555418830.043200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "Nightly completed",
        "ts": "1555428183.043300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 15:23:43,282 INFO nightly.py took 4:56:45.524303\n```",
        "ts": "1555428184.043400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 20:30:12,477 INFO archidekt.py took 0:28:18.885471\n```",
        "ts": "1555446573.043500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 20:33:25,307 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/contrib\/pyopenssl.py\", line 294, in recv_into\n    return self.connection.recv_into(*args, **kwargs)\n  File \"\/usr\/lib\/python3\/dist-packages\/OpenSSL\/SSL.py\", line 1625, in recv_into\n    self._raise_ssl_error(self._ssl, result)\n  File \"\/usr\/lib\/python3\/dist-packages\/OpenSSL\/SSL.py\", line 1448, in _raise_ssl_error\n    raise SysCallError(errno, errorcode.get(errno))\nOpenSSL.SSL.SysCallError: (104, 'ECONNRESET')\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 360, in _error_catcher\n    yield\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 666, in read_chunked\n    self._update_chunk_length()\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 598, in _update_chunk_length\n    line = self._fp.fp.readline()\n  File \"\/usr\/lib\/python3.5\/socket.py\", line 575, in readinto\n    return self._sock.recv_into(b)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/contrib\/pyopenssl.py\", line 299, in recv_into\n    raise SocketError(str(e))\nOSError: (104, 'ECONNRESET')\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/models.py\", line 750, in generate\n    for chunk in self.raw.stream(chunk_size, decode_content=True):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 490, in stream\n    for line in self.read_chunked(amt, decode_content=decode_content):\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 694, in read_chunked\n    self._original_response.close()\n  File \"\/usr\/lib\/python3.5\/contextlib.py\", line 77, in __exit__\n    self.gen.throw(type, value, traceback)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/urllib3\/response.py\", line 378, in _error_catcher\n    raise ProtocolError('Connection broken: %r' % e, e)\nurllib3.exceptions.ProtocolError: ('Connection broken: OSError(\"(104, \\'ECONNRESET\\')\",)', OSError(\"(104, 'ECONNRESET')\",))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/edhrec\/batch\/deckstats.py\", line 181, in &lt;module&gt;\n    links = get_new_decks()\n  File \"\/home\/ubuntu\/edhrec\/batch\/deckstats.py\", line 104, in get_new_decks\n    r = requests.get('<https:\/\/deckstats.net\/decks\/f\/edh-commander\/?lng=en&amp;page={}>'.format(page), headers={'User-Agent': USER_AGENT})\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/api.py\", line 75, in get\n    return request('get', url, params=params, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/api.py\", line 60, in request\n    return session.request(method=method, url=url, **kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/sessions.py\", line 686, in send\n    r.content\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/models.py\", line 828, in content\n    self._content = b''.join(self.iter_content(CONTENT_CHUNK_SIZE)) or b''\n  File \"\/usr\/local\/lib\/python3.5\/dist-packages\/requests\/models.py\", line 753, in generate\n    raise ChunkedEncodingError(e)\nrequests.exceptions.ChunkedEncodingError: ('Connection broken: OSError(\"(104, \\'ECONNRESET\\')\",)', OSError(\"(104, 'ECONNRESET')\",))\n```",
        "ts": "1555446766.043600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 21:25:40,687 INFO mtggoldfish.py took 0:31:32.055634\n```",
        "ts": "1555449901.043700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 21:47:32,560 INFO update_deckswithprices.py took 0:21:49.698164\n```",
        "ts": "1555451214.043800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-16 21:47:35,805 INFO New deck count from past 2 days: \n- Archidekt: 677\n- Other: 0\n- deckstats: 368\n- MTGGoldfish: 469\n```",
        "ts": "1555451216.043900",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 04:03:50,573 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\nAttributeError: module 'nightly' has no attribute 'make_top'\n```",
        "ts": "1555473790.044000",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 04:06:08,691 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\nAttributeError: module 'nightly' has no attribute 'make_tops'\n```",
        "ts": "1555473928.044100",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 04:32:53,708 INFO recommendations.py took 20:32:50.050786\n```",
        "ts": "1555475533.044200",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 04:55:58,539 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;stdin&gt;\", line 1, in &lt;module&gt;\n  File \"\/home\/ubuntu\/edhrec\/core\/database.py\", line 1009, in get_salt\n    saltiest = sorted(cards.get_all(), key=lambda x: x['salt'], reverse=True)[:100]\n  File \"\/home\/ubuntu\/edhrec\/core\/database.py\", line 1009, in &lt;lambda&gt;\n    saltiest = sorted(cards.get_all(), key=lambda x: x['salt'], reverse=True)[:100]\nTypeError: string indices must be integers\n```",
        "ts": "1555476918.044300",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 04:59:09,445 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"\/usr\/lib\/python3.5\/code.py\", line 91, in runcode\n    exec(code, self.locals)\n  File \"&lt;console&gt;\", line 1, in &lt;module&gt;\nNameError: name 'counts' is not defined\n```",
        "ts": "1555477109.044400",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 05:14:49,865 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\n  File \"\/home\/ubuntu\/edhrec\/web\/nightly.py\", line 33, in &lt;module&gt;\n    import edhrec.web.origin as origin\n  File \"\/home\/ubuntu\/edhrec\/web\/origin.py\", line 721\n    directory='cards'\n            ^\nSyntaxError: invalid syntax\n```",
        "ts": "1555478050.044500",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 05:15:47,675 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\n  File \"\/home\/ubuntu\/edhrec\/web\/nightly.py\", line 203, in make_top_salt\n    cardlists=origin.gettop(salt=True)\n  File \"\/home\/ubuntu\/edhrec\/web\/origin.py\", line 722, in gettop\n    ) for s in database.get_salt()]\nTypeError: 'NoneType' object is not iterable\n```",
        "ts": "1555478107.044600",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 05:16:56,070 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\n  File \"\/home\/ubuntu\/edhrec\/web\/nightly.py\", line 203, in make_top_salt\n    cardlists=origin.gettop(salt=True)\n  File \"\/home\/ubuntu\/edhrec\/web\/origin.py\", line 722, in gettop\n    ) for s in database.get_salt().values()]\nAttributeError: 'list' object has no attribute 'values'\n```",
        "ts": "1555478176.044700",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    },
    {
        "type": "message",
        "subtype": "bot_message",
        "text": "```2019-04-17 05:17:20,363 CRITICAL Uncaught exception\nTraceback (most recent call last):\n  File \"&lt;string&gt;\", line 1, in &lt;module&gt;\n  File \"\/home\/ubuntu\/edhrec\/web\/nightly.py\", line 203, in make_top_salt\n    cardlists=origin.gettop(salt=True)\n  File \"\/home\/ubuntu\/edhrec\/web\/origin.py\", line 722, in gettop\n    ) for s in database.get_salt()]\n  File \"\/home\/ubuntu\/edhrec\/web\/origin.py\", line 722, in &lt;listcomp&gt;\n    ) for s in database.get_salt()]\nKeyError: 'name'\n```",
        "ts": "1555478200.044800",
        "username": "edhrecbot",
        "icons": {
            "emoji": ":robot_face:",
            "image_64": "https:\/\/a.slack-edge.com\/37d58\/img\/emoji_2017_12_06\/apple\/1f916.png"
        },
        "bot_id": "B2867AJ5C"
    }
]